lib/current-env.js

```diff
@@ -0,0 +1,60 @@
+const process = require('node:process')
+const nodeOs = require('node:os')
+
+function isMusl (file) {
+  return file.includes('libc.musl-') || file.includes('ld-musl-')
+}
+
+function os () {
+  return process.platform
+}
+
+function cpu () {
+  return process.arch
+}
+
+function libc (osName) {
+  // this is to make it faster on non linux machines
+  if (osName !== 'linux') {
+    return undefined
+  }
+  let family
+  const report = process.report.getReport()
```

Comment:
You can disable the network flag to improve performance here.

- https://github.com/lovell/detect-libc/pull/21

Or just add a dependency on `detect-libc`.

Reply:
Looks like this works in our lowest engine setting too.


```javascript
~ $ node
Welcome to Node.js v18.17.0.
Type ".help" for more information.
> process.report.excludeNetwork = true
true
> process.report.getReport();
{
  header: {
```

Reply:
Let's make that a discrete PR so it's in the changelog, as it affects normal engines checking too, not just this new `devEngines`.  Thanks @styfle

Reply:
fwiw this will no longer run on win32 systems, which is where this is the slowest.

Reply:
FYI this is in fact affecting npm in checkPlatform, because it does a call to it in a loop
https://github.com/npm/cli/blob/780afc50e3a345feb1871a28e33fa48235bc3bd5/workspaces/arborist/lib/arborist/build-ideal-tree.js#L212

This single line is taking 26 seconds to run on my system even with the performance improvement, this is absolutely crazy and I think this should be either cached straight away when the file is imported or completely removed